src/tests/unit/test_config.py

```
@@ -0,0 +1,273 @@
+        Verifies that ConfigManager handles malformed YAML gracefully
+        and provides meaningful error messages.
+        """
+        # Create invalid YAML
+        with open(self.config_path, 'w') as f:
+            f.write("invalid: yaml: content: [")
+        
+        with patch('src.core.config.Path') as mock_path:
+            mock_path.return_value.parent.parent.parent = Path(self.temp_dir)
+            
+            # Should raise YAMLError or other parsing error
+            with self.assertRaises(Exception):
```

Ruff (0.12.2)

Do not assert blind exception: `Exception`

(B017)